cmd/gossamer/commands/root.go

```diff
@@ -0,0 +1,628 @@
+	logLevelSync    string
+	logLevelNetwork string
+	logLevelRPC     string
+	logLevelState   string
+	logLevelRuntime string
+	logLevelBABE    string
+	logLevelGRANDPA string
+
+	// Core Config
+	// role of the node. one of: full, light or authority
+	role string
+	// validator when set, the node will be an authority
+	validator bool
+
+	// Account Config
+	// key to use for the node
+	key string
+)
+
+// Flag values for persistent flags
+var (
+	// Default accounts
+	alice   bool
+	bob     bool
+	charlie bool
+
+	// Initialization flags for node
+	chain    string
+	basePath string
+)
+
+// Default values
+const (
+	// DefaultHomeEnv is the default environment variable for the base path
+	DefaultHomeEnv = "GSSMRHOME"
+)
+
+// NewRootCommand creates the root command
+func NewRootCommand() (*cobra.Command, error) {
+	cmd := &cobra.Command{
+		Use:   "gossamer",
+		Short: "Official gossamer command-line interface",
+		Long: `Gossamer is a Golang implementation of the Polkadot Host.
+Usage:
+	gossamer --chain westend-local --alice --babe-lead
+	gossamer --chain westend-dev --key alice --port 7002
+	gossamer --chain westend --key bob --port 7003
+	gossamer --chain kusama --key charlie --port 7004
+	gossamer --chain polkadot --key dave --port 7005`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return execRoot(cmd)
+		},
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			if !(cmd.Name() == "gossamer" || cmd.Name() == "init") {
+				return nil
+			}
+
+			if err := parseChainSpec(chain); err != nil {
+				return fmt.Errorf("failed to parse chain-spec: %s", err)
+			}
+
+			if err := parseBasePath(); err != nil {
+				return fmt.Errorf("failed to parse base path: %s", err)
+			}
+
+			parseAccount()
+
+			// Copy chain-spec to base-path
+			if err := copyChainSpec(config.ChainSpec, cfg.GetChainSpec(config.BasePath)); err != nil {
+				return fmt.Errorf("failed to copy chain-spec: %s", err)
+			}
+
+			if cmd.Name() == "gossamer" {
+				if err := parseRole(); err != nil {
+					return fmt.Errorf("failed to parse role: %s", err)
+				}
+
+				if err := parseTelemetryURL(); err != nil {
+					return fmt.Errorf("failed to parse telemetry-url: %s", err.Error())
+				}
+
+				if err := configureViper(config.BasePath); err != nil {
+					return fmt.Errorf("failed to configure viper: %s", err)
+				}
+
+				if err := ParseConfig(); err != nil {
+					return fmt.Errorf("failed to parse config: %s", err)
+				}
+			}
+
+			return nil
+		},
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	if err := addRootFlags(cmd); err != nil {
+		return nil, err
+	}
+
+	return cmd, nil
+}
+
+// addRootFlags adds the root flags to the command
+func addRootFlags(cmd *cobra.Command) error {
+	// global flags
+	cmd.PersistentFlags().StringVar(&basePath,
+		"base-path",
+		"",
+		"The base path for the node. Defaults to $GSSMRHOME if set")
+	cmd.PersistentFlags().StringVar(&chain,
+		"chain",
+		cfg.WestendLocalChain.String(),
```

Comment:
Is it essential to hard code it to westend here?

Reply:
No, do you feel we shouldn't have any default values?